# genes functional annotation


*environiment:* yaml.enrichment


*aim:* gain functional insight on the genes associated to crema - vicia interactions.


---


GO annotation has been carried out using 


- [PANNZER](http://ekhidna2.biocenter.helsinki.fi/sanspanz/) using no filtering and defeault parameters 


- [EggNOG-mapper](http://eggnog-mapper.embl.de/) using Insecta and Viridiplantae as Taxonomic scope and leveraging all annotations (including electronic ones).


PANNZER assigned 41331 and 99382 GO-terms respectively to crema and vicia (to 7930 and 18695 transcripts).


EggNOG-mapper assigned 391736 and 690388 GO-terms respectively to crema and vicia (to 5061 and 11897 transcripts).


Subsequent enrichment analysis legerage [topGO](https://bioconductor.org/packages/release/bioc/html/topGO.html)
and for this purpose the PANNZER and EggNOG-mapper outputs need to be reformatted using:


```
sh scripts/reformat_GOs.sh 
enrichment/PANNZER_crema/GO_crema_predictions.lst 
enrichment/eggNOG_crema/crema_eggNOG-mapper_annotations.tsv 
enrichment/GO_crema/GO_crema_geneUniverse
```


and


```
sh scripts/reformat_GOs.sh 
enrichment/PANNZER_vicia/GO_vicia_predictions.lst 
enrichment/eggNOG_vicia/vicia_eggNOG-mapper_annotations.tsv 
enrichment/GO_vicia/GO_vicia_geneUniverse
```


This script take as input:

- PANNZER output

- eggNOGmapper output

- the output file name


In total we annotated with GOterms 8615 and 18698 transcripts for crema and vicia.


Then the GO enrichment analysisis is performed using the following commands:


```
Rscript scripts/GO_enrichment.Rscript enrichment/GO_vicia/GO_vicia_geneUniverse 
abundances/vicia/vicia_deseq_gene/ BP 10 classic fisher 0.01 enrichment/GO_vicia/
```


```
Rscript scripts/GO_enrichment.Rscript enrichment/GO_crema/GO_crema_geneUniverse 
abundances/crema/crema_WGCNA_gene/ BP 10 weight fisher 0.01 enrichment/GO_crema/
```


The Rscript has several positional arguments:


1. the gene universe
2. the input folder containing the genes subsets considered
3. the onyology considered (BP/MF/CC)
4. the minimum node size
5. the algorythm
6. the statistical test
7. the p value cutoff


For this project, a hierachy-aware algoryhm (weight) was used to find enriched BP terms 
and nodes are required to have at least 10 genes associated to be 
included. Terms are considered to be enriched only if p < 0.01
in Fisher's Exact Test.


Raw GSEA output are available in:


```enrichment/GO_vicia/*tsv```


```enrichment/GO_crema/*tsv```


Then we can take a look at hub genes:


```for i in $(awk '{print $1}' abundances/crema/crema_WGCNA_gene/AD_hub_genes.tsv); do grep $i abundances/crema/crema_WGCNA_gene/AD_module_*; done```


---


GO enrichment analyses were further charchteryzed using the Rscript ```plot_GO_enrichment.Rscript```


NB: this part has been carried out from any environiment due to conflicts. 
Relevant libraries are reported inside the script and dbs can be installed 
in R using ```BiocManager::install("org.At.tair.db")``` and ```BiocManager::install("org.Dm.eg.db")```.



The Rscript positional arguments:


1. enriched goterms alog with their p value /  ```red_enrichment.tsv``` files generated by ```GO_enrichment.Rscript```
2. similarity calculation method
3. the onyology considered (BP/MF/CC)
4. treshold for finding high ranking GO terms
5. the database
6. color
7. output files
8. GO terms to prioritize in cluster

in brief:


- p values are -log10 transformed and used as scores
- calculateSimMatrix function is used to calculate score similarities between GO terms, using Rel method and org.At.tair.db db
- reduceSimMatrix is used to identify high ranking GO terms based on their semantic similarity and scores, using a treshold of 8


For vicia:


```
Rscript scripts/plot_GO_enrichment.Rscript enrichment/GO_vicia/vicia_UP_BP_red_enrichment.tsv 
Rel BP 0.6 org.At.tair.db orange images/vicia_UP.jpg GO:0015979
```
```
Rscript scripts/plot_GO_enrichment.Rscript enrichment/GO_vicia/vicia_DN_BP_red_enrichment.tsv 
Rel BP 0.6 org.At.tair.db lightblue images/vicia_DN.jpg GO:0098542,GO:0009873
```

Here are the results for vicia (upreg in orange dnreg inlightblue):


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/vicia_UP.jpg)


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/vicia_DN.jpg)


For crema, semantic clustering - leveraging D. melanogaster annotation - can be generated with:


```
Rscript scripts/plot_GO_enrichment.Rscript enrichment/GO_crema/CT_module_41_BP_red_enrichment.tsv
Rel BP 0.7 org.Dm.eg.db orange images/CT_module_41.jpg
```


just a couple of interesting modules are reported here:


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/CT_module_41.jpg)


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/AD_module_14.jpg)


The full list of clustering of trait-associated modules can be found [here]().


---


Then the genes associated to GOterms of interest were extracted;
here are some examples for:


- vicia downregulated genes associated to GO:0042136 -  neurotransmitter biosynthetic process: 


```
Rscript scripts/GO_extract.Rscript enrichment/GO_vicia/enrichment.Rdata 
GO:0042136 abundances/vicia/vicia_deseq_gene/vicia_DN_genes.lst
```


```
[1] "TRINITY_DN36133_c0_g1" "TRINITY_DN833_c0_g1" 
```


- crema genes in AD module 3 associated to GO:0042811 - pheromone biosynthetic process:


```
Rscript scripts/GO_extract.Rscript enrichment/GO_crema/enrichment.Rdata 
GO:0042811 abundances/crema/crema_WGCNA_gene/AD_module_3_genes.lst
```


```
[1] "TRINITY_DN255_c0_g1"   "TRINITY_DN52942_c0_g1"
```


Then we extracted the transcripts amminoacid sequences using:


```
for i in $(cat abundances/crema/crema_WGCNA_gene/CT_module_13_genes.lst); do 
sed -n -e "/$i/,/TRINITY/ p" annotations/crema/crema.Trinity.fasta.transdecoder.pep | head -n -1 | awk '{print $1}';
done
```


---


Additional KO annotation has been carried out using [BlastKOALA](https://www.kegg.jp/blastkoala/)
leveraging the correct Taxonomy ID and the family_eukaryotes database.
BlastKOALA assigned XXX and YYY KO-terms respectively to crema and vicia.


To further confirm the functional meaning of the observed transcriptional responses, 
we also leveraged th KO annotation generated by BlastKOALA.


```
grep -f abundances/vicia/vicia_deseq_gene/vicia_UP_genes.lst enrichment/blastKOALA_vicia/vicia_family-eukaryotes_blastKOALA.lst | grep K > enrichment/KO_vicia/vicia_UP_genes
grep -f abundances/vicia/vicia_deseq_gene/vicia_DN_genes.lst enrichment/blastKOALA_vicia/vicia_family-eukaryotes_blastKOALA.lst | grep K > enrichment/KO_vicia/vicia_DN_genes
```


and


```
for i in abundances/crema/crema_WGCNA_gene/*_module_*; 
	do 
	name=$(echo ${i::-10} | awk -F "/" '{print $NF}'); 
	grep -f $i enrichment/blastKOALA_crema/crema_family-eukaryotes_blastKOALA.lst | grep K > enrichment/KO_crema/$name; 
done
```


---


For crema a signalp analysis was carried out and then the number of protein with a secretion peptide 
was assigned to each module with:


```
for i in {1..45}; 
	do 
	secreted=$(grep -f abundances/crema/crema_WGCNA_gene/CT_module_"$i"_genes.lst enrichment/signalp_crema/crema_transdecoder_signalp.out | grep -c -w SP); 
	total=$(wc -l abundances/crema/crema_WGCNA_gene/CT_module_"$i"_genes.lst | awk '{print $1}'); 
	echo -e "module_$i\t$secreted\t$total"; 
	done > single_genes_fun/crema_signalp_CT.lst
```


and


```
for i in {1..15}; 
	do 
	secreted=$(grep -f abundances/crema/crema_WGCNA_gene/AD_module_"$i"_genes.lst enrichment/signalp_crema/crema_transdecoder_signalp.out | grep -c -w SP); 
	total=$(wc -l abundances/crema/crema_WGCNA_gene/CT_module_"$i"_genes.lst | awk '{print $1}'); 
	echo -e "module_$i\t$secreted\t$total"; 
	done > single_genes_fun/crema_signalp_AD.lst
```


The associated otputs can be found in the folder ```single_genes_fun```.


---


Furthermore, for crema we manually charachterized

- venom-associate proteins
- antimicrobial peptides
- aggression-related genes


Ddatabases were respectively created


- one based on [VenomZone (insecta)](https://venomzone.expasy.org/1413)
- one using "taxonomy:"Insecta [50557]" keyword:"Antimicrobial [KW-0929]" AND reviewed:yes" on Uniprot
- one manually curated based on the [relevant litterature](https://github.com/for-giobbe/PAINT/blob/main/enrichment/crema_aggression/aggression_db.csv)


Only the workflow for the antimicrobial peptides will be shown. After building the dbs using:


```makeblastdb -in antimicrobial_peptides.fasta -dbtype prot```


blastp searches were carried out, using:


```
blastp -query annotations/crema/crema.Trinity.fasta.transdecoder.pep 
-db enrichment/crema_antimb/antimicrobial_peptides.fasta 
-evalue 1e-7 -qcov_hsp_perc 40 -outfmt "6 qseqid sseqid evalue qcovs" 
-max_target_seqs 1 -max_hsps 1 > enrichment/crema_antimb/crema_antimb_blastp.out
```


Subsequently, the DE statistics for 


(a) the comparison between "control" and "treated" conditions and


(b) body parts


can be obtained respectively with:


```
sh scripts/extract_exp_val_crema_tissue_contrast.sh  enrichment/crema_antimb/crema_antimb_blastp.out 
> enrichment/crema_antimb/crema_antimb_expression_tissue_contrast.tsv
```


```
sh scripts/extract_exp_val_crema_treatm_contrast.sh  enrichment/crema_antimb/crema_antimb_blastp.out 
> enrichment/crema_antimb/crema_antimb_expression_treatm_contrast.tsv
```


The associated otputs can be found:


- [here](https://github.com/for-giobbe/PAINT/tree/main/enrichment/crema_venoms) for venom-associated proteins
- [here](https://github.com/for-giobbe/PAINT/tree/main/enrichment/crema_antimb) for antimicrobial peptides
- [here](https://github.com/for-giobbe/PAINT/tree/main/enrichment/crema_venoms) for aggression-related genes


---


[main](https://github.com/for-giobbe/PAINT) / 
[0](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_0.md) / 
[1](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_1.md) / 
[2](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_2.md) / 
[3](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_3.md) / 
[4](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_4.md) / 
[5](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_5.md) / 
[6](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_6.md) /
[7](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_7.md)
