## Weighted Gene Coexpression Network in crema


Here a different approach than the one used for vicia will be leveraged, due to the multiple tissues / conditions. 


---


Nonetheless the initial steps are different to those carried out for vicia. Transcript abundance can be obtained 
using the script ```slurm_abundances_crema``` which also uses bowtie and RSEM. 
Move the outputs from the main folder to the appropriate folder usings  ```mv *_rep*  abundances/crema```.


After that the expression matrices can be generated by:


```
abundance_estimates_to_matrix.pl --est_method RSEM 
--gene_trans_map assemblies/crema/crema.Trinity.fasta.gene_trans_map  --name_sample_by_basedir 
abundances/crema/A_AD_rep1/RSEM.isoforms.results 
abundances/crema/A_AD_rep2/RSEM.isoforms.results 
abundances/crema/A_AD_rep3/RSEM.isoforms.results 
abundances/crema/A_AD_rep4/RSEM.isoforms.results 
abundances/crema/A_AD_rep5/RSEM.isoforms.results 
abundances/crema/A_CT_rep1/RSEM.isoforms.results 
abundances/crema/A_CT_rep2/RSEM.isoforms.results 
abundances/crema/A_CT_rep3/RSEM.isoforms.results 
abundances/crema/A_CT_rep4/RSEM.isoforms.results 
abundances/crema/A_CT_rep5/RSEM.isoforms.results 
abundances/crema/B_AD_rep1/RSEM.isoforms.results 
abundances/crema/B_AD_rep2/RSEM.isoforms.results 
abundances/crema/B_AD_rep3/RSEM.isoforms.results 
abundances/crema/B_AD_rep4/RSEM.isoforms.results 
abundances/crema/B_AD_rep5/RSEM.isoforms.results 
abundances/crema/B_CT_rep1/RSEM.isoforms.results 
abundances/crema/B_CT_rep2/RSEM.isoforms.results 
abundances/crema/B_CT_rep3/RSEM.isoforms.results 
abundances/crema/B_CT_rep4/RSEM.isoforms.results 
abundances/crema/B_CT_rep5/RSEM.isoforms.results 
abundances/crema/C_AD_rep1/RSEM.isoforms.results 
abundances/crema/C_AD_rep2/RSEM.isoforms.results 
abundances/crema/C_AD_rep3/RSEM.isoforms.results 
abundances/crema/C_AD_rep4/RSEM.isoforms.results 
abundances/crema/C_AD_rep5/RSEM.isoforms.results 
abundances/crema/C_CT_rep1/RSEM.isoforms.results 
abundances/crema/C_CT_rep2/RSEM.isoforms.results 
abundances/crema/C_CT_rep3/RSEM.isoforms.results 
abundances/crema/C_CT_rep4/RSEM.isoforms.results 
abundances/crema/C_CT_rep5/RSEM.isoforms.results 
abundances/crema/D_AD_rep1/RSEM.isoforms.results 
abundances/crema/D_AD_rep2/RSEM.isoforms.results 
abundances/crema/D_AD_rep3/RSEM.isoforms.results 
abundances/crema/D_AD_rep4/RSEM.isoforms.results 
abundances/crema/D_AD_rep5/RSEM.isoforms.results 
abundances/crema/D_CT_rep1/RSEM.isoforms.results 
abundances/crema/D_CT_rep2/RSEM.isoforms.results 
abundances/crema/D_CT_rep3/RSEM.isoforms.results 
abundances/crema/D_CT_rep4/RSEM.isoforms.results 
abundances/crema/D_CT_rep5/RSEM.isoforms.results 
--out_prefix RSEM_crema
```


after moving the outputs to the right place with ```mv RSEM_vicia.* abundances/vicia/``` we can proceed to remove contaminants.


As before, we need to reformat the contaminants list:


```
awk -F "_" 'NF{NF-=1};1' contaminants/crema/crema.blastp.contaminants_contigs.lst | 
sed 's/ /_/g' > contaminants/crema/crema.blastp.contaminants_genes.lst
```


and then generate a raw-counts matrix without them:


```
grep -v -f contaminants/crema/crema.blastp.contaminants_genes.lst abundances/crema/RSEM_crema.gene.counts.matrix > 
abundances/crema/RSEM_crema.filtered.gene.counts.matrix
```


Then we perform an exploratory DE analysis using DESeq2:


```
run_DE_analysis.pl --matrix crema/RSEM_crema.gene.counts.matrix --samples_file ../samples_crema.txt 
--method DESeq2 --output crema_deseq_gene
```


---


Since in this experiment several conditions are present, a WGCNA approach is more suitable.
Everyting can be seamlessly performed using the command:


```
Rscript scripts/WGCNA_crema.Rscript
```


While for DE analyses we used the raw gene-counts matrix as input, WGCNA requires normalized counts
and for this purpose the TMM-normalized CPMs provided by edgeR are leveraged. Moreover - as suggested by -
we resticted the Gene Network inference to the transcripts which are consistently
expressed troughout all samples,  (27169 genes are kept


A signed network has been inferred with a power of 6, 
a dendrom cut height of 0.1 and leveraging the bidweight midcorrelation.
The trait file is and the expression matrix is the same used for the preliminary DE analysis.


The Rscript will generate several files, including:


- genes in modules positively associated to short term EFN feeding in head+thorax: ```CT_S_POS_modules_genes.lst```
- genes in modules positively associated to short term EFN feeding in the abdomen: ```AD_S_POS_modules_genes.lst```
- genes in modules positively associated to long term EFN feeding in head+thorax: ```CT_L_POS_modules_genes.lst```
- genes in modules positively associated to long term EFN feeding in the abdomen: ```AD_L_POS_modules_genes.lst```
- the top hub-genes for each module ```hub_genes.lst```
- genes present in each module


---


[prev](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_3.md) / [main](https://github.com/for-giobbe/PAINT) / [next](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_5.md)
