## Weighted Gene Coexpression Network in crema

Here a different approach than the one used for vicia will be leveraged, due to the multiple tissues / conditions. 


---


Nonetheless the initial steps are different to those carried out for vicia. Transcript abundance can be obtained 
using the script ```slurm_abundances_crema``` which also uses bowtie and RSEM. 
Move the outputs from the main folder to the appropriate folder usings  ```mv *_rep*  abundances/crema```.

After that the expression matrices can be generated by:

```
abundance_estimates_to_matrix.pl --est_method RSEM --gene_trans_map assemblies/crema/crema.Trinity.fasta.gene_trans_map  --name_sample_by_basedir 
abundances/crema/A_AD_rep1/RSEM.isoforms.results 
abundances/crema/A_AD_rep2/RSEM.isoforms.results 
abundances/crema/A_AD_rep3/RSEM.isoforms.results 
abundances/crema/A_AD_rep4/RSEM.isoforms.results 
abundances/crema/A_AD_rep5/RSEM.isoforms.results 
abundances/crema/A_CT_rep1/RSEM.isoforms.results 
abundances/crema/A_CT_rep2/RSEM.isoforms.results 
abundances/crema/A_CT_rep3/RSEM.isoforms.results 
abundances/crema/A_CT_rep4/RSEM.isoforms.results 
abundances/crema/A_CT_rep5/RSEM.isoforms.results 
abundances/crema/B_AD_rep1/RSEM.isoforms.results 
abundances/crema/B_AD_rep2/RSEM.isoforms.results 
abundances/crema/B_AD_rep3/RSEM.isoforms.results 
abundances/crema/B_AD_rep4/RSEM.isoforms.results 
abundances/crema/B_AD_rep5/RSEM.isoforms.results 
abundances/crema/B_CT_rep1/RSEM.isoforms.results 
abundances/crema/B_CT_rep2/RSEM.isoforms.results 
abundances/crema/B_CT_rep3/RSEM.isoforms.results 
abundances/crema/B_CT_rep4/RSEM.isoforms.results 
abundances/crema/B_CT_rep5/RSEM.isoforms.results 
abundances/crema/C_AD_rep1/RSEM.isoforms.results 
abundances/crema/C_AD_rep2/RSEM.isoforms.results 
abundances/crema/C_AD_rep3/RSEM.isoforms.results 
abundances/crema/C_AD_rep4/RSEM.isoforms.results 
abundances/crema/C_AD_rep5/RSEM.isoforms.results 
abundances/crema/C_CT_rep1/RSEM.isoforms.results 
abundances/crema/C_CT_rep2/RSEM.isoforms.results 
abundances/crema/C_CT_rep3/RSEM.isoforms.results 
abundances/crema/C_CT_rep4/RSEM.isoforms.results 
abundances/crema/C_CT_rep5/RSEM.isoforms.results 
abundances/crema/D_AD_rep1/RSEM.isoforms.results 
abundances/crema/D_AD_rep2/RSEM.isoforms.results 
abundances/crema/D_AD_rep3/RSEM.isoforms.results 
abundances/crema/D_AD_rep4/RSEM.isoforms.results 
abundances/crema/D_AD_rep5/RSEM.isoforms.results 
abundances/crema/D_CT_rep1/RSEM.isoforms.results 
abundances/crema/D_CT_rep2/RSEM.isoforms.results 
abundances/crema/D_CT_rep3/RSEM.isoforms.results 
abundances/crema/D_CT_rep4/RSEM.isoforms.results 
abundances/crema/D_CT_rep5/RSEM.isoforms.results 
--out_prefix RSEM_crema
```

after moving the outputs to the right place with ```mv RSEM_vicia.* abundances/vicia/``` we can proceed to remove contaminants:

first we need to reformat the contaminants list:

```
awk -F "_" 'NF{NF-=1};1' contaminants/crema/crema.blastp.contaminants_contigs.lst | sed 's/ /_/g' > contaminants/crema/crema.blastp.contaminants_genes.lst
```

and then generate a raw-counts matrix without them:

```
grep -v -f contaminants/crema/crema.blastp.contaminants_genes.lst abundances/crema/RSEM_crema.gene.counts.matrix > abundances/crema/RSEM_crema.filtered.gene.counts.matrix
```

Then we perform an exploratory DE analysis using DESeq2:

```
run_DE_analysis.pl --matrix crema/RSEM_crema.gene.counts.matrix --samples_file ../samples_crema.txt --method DESeq2 --output crema_deseq_gene
```


---



*NB:* while for DE analyses we used the raw gene-counts matrix as input, WGCNA requires normalized counts
and for this purpose we will leverage TMM-normalized TPMs.  

```
library(edgeR)

> GenewiseCounts <- read.delim("GSE60450_Lactation-GenewiseCounts.txt.gz", row.names="EntrezGeneID")
colnames(GenewiseCounts) <- substring(colnames(GenewiseCounts),1,7)

> y <- DGEList(GenewiseCounts[,-1], group=group, genes=GenewiseCounts[,1,drop=FALSE])
options(digits=3)

y <- calcNormFactors(y)
y$samples
norm_counts <- cpm(y)
```

```
#/ make the DGEList:
y <- DGEList(...)

#/ calculate TMM normalization factors:
y <- calcNormFactors(y)

#/ get the normalized counts:
cpms <- cpm(y, log=FALSE)
```

```
library(edgeR)

rnaseqMatrix = read.table("RSEM_crema.isoform.TPM.not_cross_norm", header=T, row.names=1, com='', check.names=F)
rnaseqMatrix = as.matrix(rnaseqMatrix)
rnaseqMatrix = round(rnaseqMatrix)
exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))
exp_study = calcNormFactors(exp_study)
exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors
write.table(exp_study$samples, file="RSEM_crema.isoform.TPM.not_cross_norm.TMM_info.txt", quote=F, sep="\t", row.names=F)
```

---


[prev](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_3.md) / [main](https://github.com/for-giobbe/PAINT) / [next](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_5.md)
