# expression analyses in crema


*environiment:* yaml.DE


*aim:* retrieve genes expression changes associated to EFN feeding 


---


The initial steps are the same as before.


Transcript abundance can be obtained 
using the script ```slurm_abundances_crema``` which also uses bowtie and RSEM. 


Move the outputs from the main folder to the appropriate folder usings  ```mv *_rep*  abundances/crema```.


After that the expression matrices can be generated by:


```
abundance_estimates_to_matrix.pl --est_method RSEM 
--gene_trans_map assemblies/crema/crema.Trinity.fasta.gene_trans_map  --name_sample_by_basedir 
abundances/crema/A_AD_rep1/RSEM.isoforms.results 
abundances/crema/A_AD_rep2/RSEM.isoforms.results 
abundances/crema/A_AD_rep3/RSEM.isoforms.results 
abundances/crema/A_AD_rep4/RSEM.isoforms.results 
abundances/crema/A_AD_rep5/RSEM.isoforms.results 
abundances/crema/A_CT_rep1/RSEM.isoforms.results 
abundances/crema/A_CT_rep2/RSEM.isoforms.results 
abundances/crema/A_CT_rep3/RSEM.isoforms.results 
abundances/crema/A_CT_rep4/RSEM.isoforms.results 
abundances/crema/A_CT_rep5/RSEM.isoforms.results 
abundances/crema/B_AD_rep1/RSEM.isoforms.results 
abundances/crema/B_AD_rep2/RSEM.isoforms.results 
abundances/crema/B_AD_rep3/RSEM.isoforms.results 
abundances/crema/B_AD_rep4/RSEM.isoforms.results 
abundances/crema/B_AD_rep5/RSEM.isoforms.results 
abundances/crema/B_CT_rep1/RSEM.isoforms.results 
abundances/crema/B_CT_rep2/RSEM.isoforms.results 
abundances/crema/B_CT_rep3/RSEM.isoforms.results 
abundances/crema/B_CT_rep4/RSEM.isoforms.results 
abundances/crema/B_CT_rep5/RSEM.isoforms.results 
abundances/crema/C_AD_rep1/RSEM.isoforms.results 
abundances/crema/C_AD_rep2/RSEM.isoforms.results 
abundances/crema/C_AD_rep3/RSEM.isoforms.results 
abundances/crema/C_AD_rep4/RSEM.isoforms.results 
abundances/crema/C_AD_rep5/RSEM.isoforms.results 
abundances/crema/C_CT_rep1/RSEM.isoforms.results 
abundances/crema/C_CT_rep2/RSEM.isoforms.results 
abundances/crema/C_CT_rep3/RSEM.isoforms.results 
abundances/crema/C_CT_rep4/RSEM.isoforms.results 
abundances/crema/C_CT_rep5/RSEM.isoforms.results 
abundances/crema/D_AD_rep1/RSEM.isoforms.results 
abundances/crema/D_AD_rep2/RSEM.isoforms.results 
abundances/crema/D_AD_rep3/RSEM.isoforms.results 
abundances/crema/D_AD_rep4/RSEM.isoforms.results 
abundances/crema/D_AD_rep5/RSEM.isoforms.results 
abundances/crema/D_CT_rep1/RSEM.isoforms.results 
abundances/crema/D_CT_rep2/RSEM.isoforms.results 
abundances/crema/D_CT_rep3/RSEM.isoforms.results 
abundances/crema/D_CT_rep4/RSEM.isoforms.results 
abundances/crema/D_CT_rep5/RSEM.isoforms.results 
--out_prefix RSEM_crema
```


after moving the outputs to the right place with ```mv RSEM_vicia.* abundances/vicia/``` we can proceed to remove contaminants.


As before, we need to reformat the contaminants list:


```
awk -F "_" 'NF{NF-=1};1' contaminants/crema/crema.blastp.contaminants_contigs.lst | 
sed 's/ /_/g' > contaminants/crema/crema.blastp.contaminants_genes.lst
```


Generate a raw-counts matrix without them:


```
grep -v -f contaminants/crema/crema.blastp.contaminants_genes.lst abundances/crema/RSEM_crema.gene.counts.matrix > 
abundances/crema/RSEM_crema.filtered.gene.counts.matrix
```


We can perform a DE analysis using DESeq2, with:


```
run_DE_analysis.pl --matrix crema/RSEM_crema.gene.counts.matrix --samples_file ../samples_crema.txt 
--method DESeq2 --output crema_deseq_gene
```


Then we can find all gene genes with a LogFC >+1 or LogFC <-1 and an adjusted p value < 0.01, using:  


```
Rscript scripts/DE_genes.Rscript 0.01 1 
abundances/crema/crema_deseq2_gene/RSEM_crema.filtered.gene.counts.matrix.A_CT_vs_B_CT.DESeq2.DE_results 
abundances/crema/crema_deseq2_gene/crema_A_CT_vs_B_CT_UP_genes.lst 
abundances/crema/crema_deseq2_gene/crema_A_CT_vs_B_CT_DN_genes.lst 
images/crema_A_CT_vs_B_CT_DE.jpg TRUE
```


```
Rscript scripts/DE_genes.Rscript 0.01 1 
abundances/crema/crema_deseq2_gene/RSEM_crema.filtered.gene.counts.matrix.A_CT_vs_C_CT.DESeq2.DE_results 
abundances/crema/crema_deseq2_gene/crema_A_CT_vs_C_CT_UP_genes.lst 
abundances/crema/crema_deseq2_gene/crema_A_CT_vs_C_CT_DN_genes.lst 
images/crema_A_CT_vs_C_CT_DE.jpg TRUE
```


```
Rscript scripts/DE_genes.Rscript 0.01 1 
abundances/crema/crema_deseq2_gene/RSEM_crema.filtered.gene.counts.matrix.A_CT_vs_D_CT.DESeq2.DE_results 
abundances/crema/crema_deseq2_gene/crema_A_CT_vs_D_CT_UP_genes.lst 
abundances/crema/crema_deseq2_gene/crema_A_CT_vs_D_CT_DN_genes.lst 
images/crema_A_CT_vs_D_CT_DE.jpg TRUE
```


```
Rscript scripts/DE_genes.Rscript 0.01 1 
abundances/crema/crema_deseq2_gene/RSEM_crema.filtered.gene.counts.matrix.A_AD_vs_B_AD.DESeq2.DE_results 
abundances/crema/crema_deseq2_gene/crema_A_AD_vs_B_AD_UP_genes.lst 
abundances/crema/crema_deseq2_gene/crema_A_AD_vs_B_AD_DN_genes.lst 
images/crema_A_AD_vs_B_AD_DE.jpg TRUE
```


```
Rscript scripts/DE_genes.Rscript 0.01 1 
abundances/crema/crema_deseq2_gene/RSEM_crema.filtered.gene.counts.matrix.A_AD_vs_C_AD.DESeq2.DE_results 
abundances/crema/crema_deseq2_gene/crema_A_AD_vs_C_AD_UP_genes.lst 
abundances/crema/crema_deseq2_gene/crema_A_AD_vs_C_AD_DN_genes.lst 
images/crema_A_AD_vs_C_AD_DE.jpg TRUE
```


```
Rscript scripts/DE_genes.Rscript 0.01 1 
abundances/crema/crema_deseq2_gene/RSEM_crema.filtered.gene.counts.matrix.A_AD_vs_D_AD.DESeq2.DE_results 
abundances/crema/crema_deseq2_gene/crema_A_AD_vs_D_AD_UP_genes.lst 
abundances/crema/crema_deseq2_gene/crema_A_AD_vs_D_AD_DN_genes.lst 
images/crema_A_AD_vs_D_AD_DE.jpg TRUE
```


And here are the number of DE genes:


```
 wc -l abundances/crema/crema_deseq2_gene/*_genes.lst
      150 abundances/crema/crema_deseq2_gene/crema_A_AD_vs_B_AD_DN_genes.lst
      352 abundances/crema/crema_deseq2_gene/crema_A_AD_vs_B_AD_UP_genes.lst
       81 abundances/crema/crema_deseq2_gene/crema_A_AD_vs_C_AD_DN_genes.lst
      262 abundances/crema/crema_deseq2_gene/crema_A_AD_vs_C_AD_UP_genes.lst
      119 abundances/crema/crema_deseq2_gene/crema_A_AD_vs_D_AD_DN_genes.lst
      174 abundances/crema/crema_deseq2_gene/crema_A_AD_vs_D_AD_UP_genes.lst
       69 abundances/crema/crema_deseq2_gene/crema_A_CT_vs_B_CT_DN_genes.lst
      172 abundances/crema/crema_deseq2_gene/crema_A_CT_vs_B_CT_UP_genes.lst
       50 abundances/crema/crema_deseq2_gene/crema_A_CT_vs_C_CT_DN_genes.lst
      270 abundances/crema/crema_deseq2_gene/crema_A_CT_vs_C_CT_UP_genes.lst
       30 abundances/crema/crema_deseq2_gene/crema_A_CT_vs_D_CT_DN_genes.lst
      192 abundances/crema/crema_deseq2_gene/crema_A_CT_vs_D_CT_UP_genes.lst
```


---


Also a WGCNA analysisi was performed on crema, which can be found [here](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_4b.md)


---


[main](https://github.com/for-giobbe/PAINT) / 
[0](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_0.md) / 
[1](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_1.md) / 
[2](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_2.md) / 
[3](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_3.md) / 
[4](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_4.md) / 
[5](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_5.md) / 
[6](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_6.md) /
[7](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_7.md)
