## Weighted Gene Coexpression Network in crema


**environiment:** yaml.WGCNA

**aim:** retrieve genes associate to short-term and long-term EFN feeding with a WGCN approach


---


A different approach than the one used for vicia is used,
yet the initial steps are the same as before.


Transcript abundance can be obtained 
using the script ```slurm_abundances_crema``` which also uses bowtie and RSEM. 

Move the outputs from the main folder to the appropriate folder usings  ```mv *_rep*  abundances/crema```.


After that the expression matrices can be generated by:


```
abundance_estimates_to_matrix.pl --est_method RSEM 
--gene_trans_map assemblies/crema/crema.Trinity.fasta.gene_trans_map  --name_sample_by_basedir 
abundances/crema/A_AD_rep1/RSEM.isoforms.results 
abundances/crema/A_AD_rep2/RSEM.isoforms.results 
abundances/crema/A_AD_rep3/RSEM.isoforms.results 
abundances/crema/A_AD_rep4/RSEM.isoforms.results 
abundances/crema/A_AD_rep5/RSEM.isoforms.results 
abundances/crema/A_CT_rep1/RSEM.isoforms.results 
abundances/crema/A_CT_rep2/RSEM.isoforms.results 
abundances/crema/A_CT_rep3/RSEM.isoforms.results 
abundances/crema/A_CT_rep4/RSEM.isoforms.results 
abundances/crema/A_CT_rep5/RSEM.isoforms.results 
abundances/crema/B_AD_rep1/RSEM.isoforms.results 
abundances/crema/B_AD_rep2/RSEM.isoforms.results 
abundances/crema/B_AD_rep3/RSEM.isoforms.results 
abundances/crema/B_AD_rep4/RSEM.isoforms.results 
abundances/crema/B_AD_rep5/RSEM.isoforms.results 
abundances/crema/B_CT_rep1/RSEM.isoforms.results 
abundances/crema/B_CT_rep2/RSEM.isoforms.results 
abundances/crema/B_CT_rep3/RSEM.isoforms.results 
abundances/crema/B_CT_rep4/RSEM.isoforms.results 
abundances/crema/B_CT_rep5/RSEM.isoforms.results 
abundances/crema/C_AD_rep1/RSEM.isoforms.results 
abundances/crema/C_AD_rep2/RSEM.isoforms.results 
abundances/crema/C_AD_rep3/RSEM.isoforms.results 
abundances/crema/C_AD_rep4/RSEM.isoforms.results 
abundances/crema/C_AD_rep5/RSEM.isoforms.results 
abundances/crema/C_CT_rep1/RSEM.isoforms.results 
abundances/crema/C_CT_rep2/RSEM.isoforms.results 
abundances/crema/C_CT_rep3/RSEM.isoforms.results 
abundances/crema/C_CT_rep4/RSEM.isoforms.results 
abundances/crema/C_CT_rep5/RSEM.isoforms.results 
abundances/crema/D_AD_rep1/RSEM.isoforms.results 
abundances/crema/D_AD_rep2/RSEM.isoforms.results 
abundances/crema/D_AD_rep3/RSEM.isoforms.results 
abundances/crema/D_AD_rep4/RSEM.isoforms.results 
abundances/crema/D_AD_rep5/RSEM.isoforms.results 
abundances/crema/D_CT_rep1/RSEM.isoforms.results 
abundances/crema/D_CT_rep2/RSEM.isoforms.results 
abundances/crema/D_CT_rep3/RSEM.isoforms.results 
abundances/crema/D_CT_rep4/RSEM.isoforms.results 
abundances/crema/D_CT_rep5/RSEM.isoforms.results 
--out_prefix RSEM_crema
```


after moving the outputs to the right place with ```mv RSEM_vicia.* abundances/vicia/``` we can proceed to remove contaminants.


As before, we need to reformat the contaminants list:


```
awk -F "_" 'NF{NF-=1};1' contaminants/crema/crema.blastp.contaminants_contigs.lst | 
sed 's/ /_/g' > contaminants/crema/crema.blastp.contaminants_genes.lst
```


Generate a raw-counts matrix without them:


```
grep -v -f contaminants/crema/crema.blastp.contaminants_genes.lst abundances/crema/RSEM_crema.gene.counts.matrix > 
abundances/crema/RSEM_crema.filtered.gene.counts.matrix
```


We can perform an exploratory DE analysis using DESeq2, which is indeed unpractical due to the many conditions:


```
run_DE_analysis.pl --matrix crema/RSEM_crema.gene.counts.matrix --samples_file ../samples_crema.txt 
--method DESeq2 --output crema_deseq_gene
```


---


Since in this experiment several conditions are present, a WGCNA approach is more suitable.
Everyting can be seamlessly performed using the command:


```
Rscript scripts/WGCNA.Rscript 6 signed 50 0.1 0.6 bicor 
abundances/crema/RSEM_crema.filtered.gene.counts.matrix abundances/crema/crema_WGCNA_gene/crema_traits
```


While for DE analyses we used the raw gene-counts matrix as input, WGCNA requires normalized counts
and for this purpose the TMM-normalized CPMs provided by edgeR are generated by the script. 
Moreover - as suggested by authors [here](https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html) -
we resticted the Gene Network inference to the transcripts which are consistently
expressed troughout all samples (27169).


Furthermore, the script includes several parameters:


- soft-thresholding power for network construction
- network type
- minimum module size for module detection
- dendrogram cut height for module merging
- minimum eigengene coinnectivtivty for a module not to be disbanded
- correlation used

A signed network has been inferred with a power of 6, 
a dendrom cut height of 0.1 and leveraging the bidweight midcorrelation.
The expression matrix is the same used for the preliminary DE analysis and here is the trait file:


```
sed 's/,/ /g' abundances/crema/crema_WGCNA_gene/crema_traits | sed "s/sample/n \tsample/g" | column -t
```


```
n   sample     AD_N  AD_S  AD_L  CT_N  CT_S  CT_L
01  A_AD_rep1  1     0     0     0     0     0
02  A_AD_rep2  1     0     0     0     0     0
03  A_AD_rep3  1     0     0     0     0     0
04  A_AD_rep4  1     0     0     0     0     0
05  A_AD_rep5  1     0     0     0     0     0
06  A_CT_rep1  0     0     0     1     0     0
07  A_CT_rep2  0     0     0     1     0     0
08  A_CT_rep3  0     0     0     1     0     0
09  A_CT_rep4  0     0     0     1     0     0
10  A_CT_rep5  0     0     0     1     0     0
11  B_AD_rep1  0     0     1     0     0     0
12  B_AD_rep2  0     0     1     0     0     0
13  B_AD_rep3  0     0     1     0     0     0
14  B_AD_rep4  0     0     1     0     0     0
15  B_AD_rep5  0     0     1     0     0     0
16  B_CT_rep1  0     0     0     0     0     1
17  B_CT_rep2  0     0     0     0     0     1
18  B_CT_rep3  0     0     0     0     0     1
19  B_CT_rep4  0     0     0     0     0     1
20  B_CT_rep5  0     0     0     0     0     1
21  C_AD_rep1  0     1     0     0     0     0
22  C_AD_rep2  0     1     0     0     0     0
23  C_AD_rep3  0     1     0     0     0     0
24  C_AD_rep4  0     1     0     0     0     0
25  C_AD_rep5  0     1     0     0     0     0
26  C_CT_rep1  0     0     0     0     1     0
27  C_CT_rep2  0     0     0     0     1     0
28  C_CT_rep3  0     0     0     0     1     0
29  C_CT_rep4  0     0     0     0     1     0
30  C_CT_rep5  0     0     0     0     1     0
31  D_AD_rep1  0     1     1     0     0     0
32  D_AD_rep2  0     1     1     0     0     0
33  D_AD_rep3  0     1     1     0     0     0
34  D_AD_rep4  0     1     1     0     0     0
35  D_AD_rep5  0     1     1     0     0     0
36  D_CT_rep1  0     0     0     0     1     1
37  D_CT_rep2  0     0     0     0     1     1
38  D_CT_rep3  0     0     0     0     1     1
39  D_CT_rep4  0     0     0     0     1     1
40  D_CT_rep5  0     0     0     0     1     1
```


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/crema_WGCNA_custom_heatmap.jpg)


The Rscript will generate several files, including:


- genes in modules positively associated to short term EFN feeding in head+thorax: ```CT_S_POS_modules_genes.lst```
- genes in modules positively associated to short term EFN feeding in the abdomen: ```AD_S_POS_modules_genes.lst```
- genes in modules positively associated to long term EFN feeding in head+thorax: ```CT_L_POS_modules_genes.lst```
- genes in modules positively associated to long term EFN feeding in the abdomen: ```AD_L_POS_modules_genes.lst```
- the top hub-genes for each module ```hub_genes.lst```
- genes present in each module


---


[prev](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_3.md) / [main](https://github.com/for-giobbe/PAINT) / [next](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_5.md)
