# GO/KO annotation and enrichment analyses


*environiment:* yaml.enrichment


*aim:* gain functional insight on the genes associated to crema - vicia interactions.


---


GO annotation has been carried out using [PANNZER](http://ekhidna2.biocenter.helsinki.fi/sanspanz/)
using no filtering and defeault parameters. 
PANNZER assigned 39826 and 99954 GO-terms respectively to crema and vicia (7526 and 18794 transcripts).


Subsequent enrichment analysis legerage [topGO](https://bioconductor.org/packages/release/bioc/html/topGO.html)
and for this purpose the PANNZER output needs to be reformatted using:


```
sh scripts/reformat_GOs.sh enrichment/GO_crema/GO_crema_predictions.lst enrichment/GO_crema/GO_crema_geneUniverse
```


and


```
sh scripts/reformat_GOs.sh enrichment/GO_vicia/GO_vicia_predictions.lst enrichment/GO_vicia/GO_vicia_geneUniverse
```


The the GO enrichment analysisis is performed using the following commands:


```
Rscript scripts/GO_enrichment.Rscript enrichment/GO_vicia/GO_vicia_geneUniverse 
abundances/vicia/vicia_deseq_gene/ BP 10 elim fisher 0.05 enrichment/GO_vicia/
```


and


```
Rscript scripts/GO_enrichment.Rscript enrichment/GO_crema/GO_crema_geneUniverse 
abundances/vicia/crema_WGCNA_gene/ BP 10 weight fisher 0.05 enrichment/GO_crema/
```


The Rscript has several positional arguments:


1. the gene universe
2. the input folder containing the genes subsets considered
3. the onyology considered (BP/MF/CC)
4. the minimum node size
5. the algorythm
6. the statistical test
7. the p value cutoff


For this project, a hierachy-aware algoryhm (weight) was used to find enriched BP terms 
and nodes are required to have at least 3 genes associated to be 
included. Terms are considered to be enriched only if p < 0.05
in Fisher's Exact Test.


Raw GSEA output are available in:


```enrichment/GO_vicia/*tsv```


```enrichment/GO_crema/*tsv```


---


GO enrichment analyses were further charchteryzed with [rrvgo](http://www.bioconductor.org/packages/release/bioc/vignettes/rrvgo/inst/doc/rrvgo.html), 
using the Rscript ```plot_GO_enrichment.Rscript```


NB: this part has been carried out from any environiment due to conflicts. 
Relevant libraries are reported inside the script and dbs can be installed 
in R using ```BiocManager::install("org.At.tair.db")``` and ```BiocManager::install("org.Dm.eg.db")```.



The Rscript positional arguments:


1. enriched goterms alog with their p value /  ```red_enrichment.tsv``` files generated by ```GO_enrichment.Rscript```
2. similarity calculation method
3. the onyology considered (BP/MF/CC)
4. treshold for finding high ranking GO terms
5. the database
6. color
7. output files


in brief:


- p values are -log10 transformed and used as scores
- calculateSimMatrix function is used to calculate score similarities between GO terms, using Rel method and org.At.tair.db db
- reduceSimMatrix is used to identify high ranking GO terms based on their semantic similarity and scores, using a treshold of 8


For vicia:


```
Rscript scripts/enrichment_vicia_def2.R enrichment/GO_vicia/vicia_UP_BP_red_enrichment.tsv 
Rel BP 0.7 org.At.tair.db orange images/vicia_UP.jpg
```
```
Rscript scripts/enrichment_vicia_def2.R enrichment/GO_vicia/vicia_DN_BP_red_enrichment.tsv 
Rel BP 0.7 org.At.tair.db lightblue images/vicia_DN.jpg
```

Here are the results for vicia (upreg in orange dnreg inlightblue):


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/vicia_UP.jpg)


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/vicia_DN.jpg)


For crema, semantic clustering - leveraging D. melanogaster annotation - can be generated with:


```
for i in {1..45}; do
Rscript scripts/scripts/plot_GO_enrichment.Rscript  enrichment/GO_crema/AD_module_$i_BP_red_enrichment.tsv
Rel BP 0.7 org.At.tair.db orange images/vicia_UP.jpg;
done
```


just a couple of interesting modules are reported here:


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/vicia_UP.jpg)


![Image description](https://github.com/for-giobbe/PAINT/blob/main/images/vicia_UP.jpg)


The full list of clustering of trait-associated modules can be found [here].



---


Then the genes associated to GOterms of interest were extracted;
here are some examples for:


- vicia downregulated genes associated to GO:0042136 -  neurotransmitter biosynthetic process: 


```
Rscript scripts/GO_extract.Rscript enrichment/GO_vicia/enrichment.Rdata 
GO:0042136 abundances/vicia/vicia_deseq_gene/vicia_DN_genes.lst
```


```
[1] "TRINITY_DN36133_c0_g1" "TRINITY_DN833_c0_g1" 
```
The first is a homolog to serine hydroxymethyltransferase, the second to inducible nitrate reductase. 


- crema genes in AD module 3 associated to GO:0042811 - pheromone biosynthetic process:


```
Rscript scripts/GO_extract.Rscript enrichment/GO_crema/enrichment.Rdata 
GO:0042811 abundances/crema/crema_WGCNA_gene/AD_module_3_genes.lst
```


```
[1] "TRINITY_DN255_c0_g1"   "TRINITY_DN52942_c0_g1"
```


---


KO annotation has been carried out using [BlastKOALA](https://www.kegg.jp/blastkoala/)
leveraging the correct Taxonomy ID and the family_eukaryotes database.
BlastKOALA assigned XXX and YYY KO-terms respectively to crema and vicia.


Subsequent enrichment analysis legeraged [KEGGREST](https://www.bioconductor.org/packages/release/bioc/html/KEGGREST.html)


---


[main](https://github.com/for-giobbe/PAINT) / 
[0](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_0.md) / 
[1](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_1.md) / 
[2](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_2.md) / 
[3](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_3.md) / 
[4](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_4.md) / 
[5](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_5.md) / 
[6](https://github.com/for-giobbe/PAINT/blob/main/markdowns/part_6.md)
